### PR TITLE
Unsaved changes from min commission / max voting power tests etc.

### PR DESCRIPTION
**Pull request by jzvikart**

### CODE REVIEW

**Review by khdegraaf — APPROVED** [review]

File: test/integration/src/features/margin.py [hunk 1]
```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
             assert True  # no-op line just for setting a breakpoint
 
     def test_swap(self):
-        self.test_swap_rowan_to_external()
+        # self.test_swap_rowan_to_external()
         # self.test_swap_external_to_external()
+        self.test_margin()
+
+    def test_margin(self):
```
Comment (smartyalgo) [inline test/integration/src/features/margin.py:176]: Can the naming be more descriptive to what it's testing?

File: test/integration/src/features/margin.py [hunk 2]
```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
...
+
+        with self.with_test_env(self.default_pool_setup) as (env, sifnoded, pools_definitions):
+            account = sifnoded.create_addr()
+            env.fund(account, {
+                ROWAN: 10**25,
+                collateral_asset: 10**25,
+            })
+            margin_params = sifnoded.query_margin_params()
+
+            # sifnoded.tx_margin_whitelist(env.clp_admin, account, broadcast_mode="block")
```
Comment (smartyalgo) [inline test/integration/src/features/margin.py:190]: Remove commented code?

File: test/integration/src/features/margin.py [hunk 3]
```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
...
+
+            pool_before_open = sifnoded.query_pools_sorted()[collateral_asset]
+            balance_before_open = sifnoded.get_balance(account)
+            mtp_positions_before_open = sifnoded.query_margin_positions_for_address(account)
+            res = sifnoded.margin_open_simple(account, borrow_asset, collateral_asset=collateral_asset,
```
Comment (smartyalgo) [inline test/integration/src/features/margin.py:195]: Not needed for this PR, what is the behaviour when this tx errors out?

File: test/integration/src/features/margin.py [hunk 4]
```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
...
+            account = sifnoded.create_addr()
+            env.fund(account, {
+                ROWAN: 10**25,
+                collateral_asset: 10**25,
+            })
+            margin_params = sifnoded.query_margin_params()
+
+            # sifnoded.tx_margin_whitelist(env.clp_admin, account, broadcast_mode="block")
+
+            pool_before_open = sifnoded.query_pools_sorted()[collateral_asset]
```
Comment (smartyalgo) [inline test/integration/src/features/margin.py:192]: variable is never used. I think we can make assertions against `pool_after_open` ?

File: test/integration/src/features/margin.py [hunk 5]
```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
...
+            mtp_positions_before_open = sifnoded.query_margin_positions_for_address(account)
+            res = sifnoded.margin_open_simple(account, borrow_asset, collateral_asset=collateral_asset,
+                collateral_amount=collateral_amount, leverage=leverage, position="long")
+            mtp_id = int(res["id"])
+            pool_after_open = sifnoded.query_pools_sorted()[collateral_asset]
+            balance_after_open = sifnoded.get_balance(account)
+            mtp_positions_after_open = sifnoded.query_margin_positions_for_address(account)
+
+            assert len(mtp_positions_before_open) == 0
+            assert len(mtp_positions_after_open) == 1
```
Comment (smartyalgo) [inline test/integration/src/features/margin.py:203]: Nit: Better to assert == (before_open + 1) ?

File: test/integration/src/features/margin.py [hunk 6]
```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
...
+
+            open_borrow_delta = balance_after_open.get(borrow_asset, 0) - balance_before_open.get(borrow_asset, 0)
```
Comment (smartyalgo) [inline test/integration/src/features/margin.py:205]: Missing assertion on variable

**Pull request merged**